dist/tools/ethos/ethos.c

```diff
@@ -49,8 +52,18 @@ static void usage(void)
 
 static void checked_write(int handle, void *buffer, int nbyte)
 {
-    if (write(handle, buffer, nbyte) != nbyte) {
-        fprintf(stderr, "write to fd %i failed: %s\n", handle, strerror(errno));
+    ssize_t res;
+    while (nbyte > 0) {
+        res = write(handle, buffer, nbyte);
+        if (res < 0) {
```

Comment:
Accidentally left over from debugging. I was afraid that the write would return 0 sometimes if the device driver's buffer is full, but it seems I was wrong, I have not seen a zero write happen yet.